pygeos/io.py

```diff
@@ -172,3 +175,38 @@ def from_wkb(geometry, **kwargs):
     # of array elements)
     geometry = np.asarray(geometry, dtype=object)
     return lib.from_wkb(geometry, **kwargs)
+
+
+def from_shapely(geometry, **kwargs):
+    """Creates geometries from shapely Geometry objects.
+
+    This function requires the GEOS version of PyGEOS and shapely to be equal.
+
+    Parameters
+    ----------
+    geometry : shapely Geometry object or array_like
+
+    Examples
+    --------
+    >>> from shapely.geometry import Point   # doctest: +SKIP
+    >>> from_wkt(Point(1, 2))   # doctest: +SKIP
+    <pygeos.Geometry POINT (1 2)>
+    """
+    from shapely.geos import geos_version_string as shapely_geos_version
```

Comment:
Shapely also has the `shapely._buildcfg.geos_library_path` which might make it possible to check that both are using the exact same library. But the same version should be enough? (I am no expert in this, I don't know if it possible that if the same version is built with a different C toolchain, that this can still give binary incompatibilities)

Reply:
Maybe, I am also no expert on this. Accessing private shapely attributes does have its own problems.

Reply:
One question to approach this issue: how to get the actual library path (at runtime) from our C extension?

Reply:
From shapely, it seems this:

```
In [15]: import shapely.geos                                                                                                                                                                                       

In [16]: shapely.geos._lgeos                                                                                                                                                                                       
Out[16]: <CDLL '/home/joris/miniconda3/envs/test-pygeos-shapely-wheels/lib/python3.7/site-packages/shapely/.libs/libgeos_c-bd8d3f16.so.1.10.2', handle 55af2a25e300 at 0x7f4eaa8e8dd8>
```

shows where it is coming from. Above, this is from a wheel that includes geos. 
And below is from a conda env:

```
In [25]: shapely.geos._lgeos                                                                                                                                                                                       
Out[25]: <CDLL '/home/joris/miniconda3/envs/dev/lib/libgeos_c.so.1', handle 5603028c7f00 at 0x7f714ea6cf28>
```

(but not sure how to get this path out of the repr)

But that's only for shapely of course, we don't have that for pygos